modules/token-yields/handlers/sources/rate-provider-handler.ts

```diff
@@ -0,0 +1,103 @@
+import { TokenYieldHandler } from '../../types';
+import { Chain } from '@prisma/client';
+import { prisma } from '../../../../prisma/prisma-client';
+import { getViemClient } from '../../../sources/viem-client';
+import { formatEther, parseAbiItem, zeroAddress, Hex } from 'viem';
+import { blockNumbers } from '../../../block-numbers';
+import { daysAgo } from '../../../common/time';
+
+const abi = [parseAbiItem('function getRate() view returns(uint)')];
+
+export const rateProviderHandler: TokenYieldHandler = async ({
+    chain,
+    intervalInDays = 30,
+}: {
+    chain: Chain;
+    intervalInDays: number;
+}) => {
+    try {
+        const poolTokensWithRateProvider = await prisma.prismaPoolToken.findMany({
+            where: {
+                chain,
+                AND: [
+                    {
+                        priceRateProvider: {
+                            not: null,
+                        },
+                    },
+                    {
+                        priceRateProvider: {
+                            not: zeroAddress,
+                        },
+                    },
+                    {
+                        priceRate: {
+                            not: '1.0',
+                        },
+                    },
+                    {
+                        priceRate: {
+                            not: '1',
+                        },
+                    },
```

Comment:
why do we need these filters? I believe this could actually prevent getting APRs from a new rate provider. I believe they are initialized with rate 1

Reply:
rate = 1 suggests there is no APR. Since many rate providers have set it to 1, they can be filtered out early with this condition.